src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -0,0 +1,92 @@
+  trlNonCoreMandatorySchema
+} from "./NonCoreMandatoryValidations";
+
+const validateRecordCompleteness = (validationSchema: ObjectSchema, techRecordFields: any) => {
+  return validationSchema.validate(techRecordFields, {stripUnknown: true});
+};
+
+const validateVehicleAttributes = (vehicleType: string, vehicleAttributes: any) => {
+  if (vehicleType !== VEHICLE_TYPE.TRL) {
+    return validateRecordCompleteness(psvHgvCarLgvMotoCoreMandatoryVehicleAttributes, vehicleAttributes);
+  } else {
+    return validateRecordCompleteness(trlCoreMandatoryVehicleAttributes, vehicleAttributes);
+  }
+};
+
+const validateCoreAndNonCoreMandatoryTechRecordAttributes = (vehicleType: string, techRecordWrapper: ITechRecordWrapper) => {
+  let coreMandatoryValidationResult;
+  let nonCoreMandatoryValidationResult;
+  let coreMandatorySchema: ObjectSchema | any;
+  let nonCoreMandatorySchema: ObjectSchema | any;
+  let techRecord = techRecordWrapper.techRecord[0];
+  if (vehicleType === VEHICLE_TYPE.HGV) {
+    coreMandatorySchema = hgvCoreMandatorySchema;
+    nonCoreMandatorySchema = hgvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.PSV) {
+    coreMandatorySchema = psvCoreMandatorySchema;
+    nonCoreMandatorySchema = psvNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.TRL) {
+    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
+    coreMandatorySchema = trlCoreMandatorySchema;
+    nonCoreMandatorySchema = trlNonCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.LGV) {
+    coreMandatorySchema = lgvCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.CAR) {
+    coreMandatorySchema = carCoreMandatorySchema;
+  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
+    coreMandatorySchema = motorcycleCoreMandatorySchema;
+  }
+  coreMandatoryValidationResult = coreMandatorySchema ? validateRecordCompleteness(coreMandatorySchema, techRecord) : undefined;
+  nonCoreMandatoryValidationResult = nonCoreMandatorySchema ? validateRecordCompleteness(nonCoreMandatorySchema, techRecord) : undefined;
+  return {coreMandatoryValidationResult, nonCoreMandatoryValidationResult};
+};
+
+export const computeRecordCompleteness = (techRecordWrapper: ITechRecordWrapper): string => {
+  let recordCompleteness = RECORD_COMPLETENESS_ENUM.COMPLETE;
+  let isCoreMandatoryValid = true;
+  let isNonCoreMandatoryValid = true;
+  if (!techRecordWrapper.systemNumber) {
+    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
+  }
+  const generalAttributes = {
+    systemNumber: techRecordWrapper.systemNumber,
+    vin: techRecordWrapper.vin,
+    primaryVrm: techRecordWrapper.primaryVrm,
+    trailerId: techRecordWrapper.trailerId
+  };
+  const vehicleType = techRecordWrapper.techRecord[0].vehicleType;
+  if (!vehicleType) {
+    return RECORD_COMPLETENESS_ENUM.SKELETON;
+  }
+  const vehicleAttributeValidation = validateVehicleAttributes(vehicleType, generalAttributes);
```

Comment:
I think we don't need this variable as we are only using it for if condition.

Reply:
done